### DISCUSSION TITLE
User membership of client roles don't seem to be pulled in by role-ldap-mapper

### DISCUSSION THREAD

**Discussion by fostermi** [keycloak/keycloak#31683] [Q&A]
I'm trying to configure an application that can only parse "Roles" from the access token (not "Groups) and have set up an role-ldap-mapper to grab a set of specific LDAP groups begining with say (cn=foo-*) and create client roles. This works, except that users are not showing as a member of this client role unless I specifically add them in KC. Is this not supposed to be automatic using the configured mapper "memberOf" user attribute for a user with that particular LDAP group?

**Comment by demetthyl** [comment]
Hello, we have absolutely the same issue on version 26.5.6:
* a role ldap mapper generate client roles
* from the client menus, we see members assigned to the client roles
* from the user menus, the rolemapping to client roles is null

If i assign manually every client roles to the user, then the rolemapping is OK.

Why isn't natively bi directional when client roles are generated via mapper?

Guillaume
